router_types.c

```diff
@@ -44,6 +47,55 @@ static const struct router_type *router_types[] = {
 	NULL,
 };
 
+int read_mac(uint8_t mac[ETH_ALEN], const char *macstr)
+{
+	int ret;
+
+	ret = sscanf(macstr, "%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX",
+				 &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+	if (ret != 6)
+		ret = sscanf(macstr, "%02hhX-%02hhX-%02hhX-%02hhX-%02hhX-%02hhX",
+					 &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+
+	return (ret == 6);
+}
+
+void mac_whitelist_add(char *macstr)
+{
+	uint8_t mac[ETH_ALEN];
```

Comment:
Please add an empty line between the variables and the code. There are a lot of other similar problems in the changes